### PR TITLE
Add WeChat.app v2.3.14

### PR DESCRIPTION
**Pull request by zzzhouuu**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

**Comment by commitay** [comment]
Ping @vitorgalvao 

This was removed a while ago. https://github.com/Homebrew/homebrew-cask/issues/10897#issuecomment-97450434

The homepage has links for direct download and MAS with the same version.

**Comment by zhouyoufa** [comment]
That's not going to happen now. #10897 

Want to have a variety of alternative ways to install so add this cask again

**Comment by vitorgalvao** [comment]
> That's not going to happen now.

What makes you say that?

Regardless, if the old reason is no longer relevant I see no issue in readding.

File: Casks/wechat.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+cask 'wechat' do
+  version '2.3.14'
```
Comment (vitorgalvao) [inline Casks/wechat.rb]: The website says `2.3.13`.

Comment (zhouyoufa) [reply Casks/wechat.rb]: The official pushed the 2.3.14 update, but the website hasn't been updated yet.

Comment (commitay) [reply Casks/wechat.rb]: homepage now has 2.3.14

**Pull request merged**